controllers/nutanixmachine_controller.go

```diff
@@ -260,19 +260,35 @@ func (r *NutanixMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	v3Client, err := getPrismCentralClientForCluster(ctx, ntxCluster, r.SecretInformer, r.ConfigMapInformer)
+	v3Client, err := getPrismCentralV3ClientForCluster(ctx, ntxCluster, r.SecretInformer, r.ConfigMapInformer)
 	if err != nil {
 		log.Error(err, "error occurred while fetching prism central client")
 		return reconcile.Result{}, err
 	}
 
 	rctx := &nctx.MachineContext{
-		Context:        ctx,
-		Cluster:        cluster,
-		Machine:        machine,
-		NutanixCluster: ntxCluster,
-		NutanixMachine: ntxMachine,
-		NutanixClient:  v3Client,
+		Context:         ctx,
+		Cluster:         cluster,
+		Machine:         machine,
+		NutanixCluster:  ntxCluster,
+		NutanixMachine:  ntxMachine,
+		NutanixClientV3: v3Client,
+	}
+
+	createV4Client, err := isPrismCentralV4Compatible(ctx, v3Client)
+	if err != nil {
+		log.Error(err, "error occurred while checking compatibility for Prism Central v4 APIs")
+	} else {
+		if createV4Client {
```

Comment:
nit: we can flatten this and remove the `else`. `createV4Client` should always be false in case of errors.